packages/contractkit/src/governance/proposals.ts

```diff
@@ -0,0 +1,87 @@
+import { keccak256 } from 'ethereumjs-util'
+import Contract from 'web3/eth/contract'
+import { Transaction, TransactionObject } from 'web3/eth/types'
+
+import { concurrentMap } from '@celo/utils/lib/async'
+
+import { AllContracts, CeloContract } from '../base'
+import { obtainKitContractDetails } from '../explorer/base'
+import { BlockExplorer, CallDetails } from '../explorer/block-explorer'
+import { ContractKit } from '../kit'
+import { CeloTransactionObject, numberLikeToString } from '../wrappers/BaseWrapper'
+import { Proposal, ProposalTransaction } from '../wrappers/Governance'
+
+export interface ProposalTransactionJSON {
+  callDetails: Pick<CallDetails, 'contract' | 'function' | 'args'>
+  value: string
+}
+
+export class ProposalUtility extends Proposal {
+  constructor(transactions: ProposalTransaction[], private readonly kit: ContractKit) {
+    super(transactions)
+  }
+
+  static fromProposalAndKit(proposal: Proposal, kit: ContractKit) {
+    return new ProposalUtility(proposal.transactions, kit)
+  }
+
+  get hash(): Buffer {
+    const paramsEncoded = this.kit.web3.eth.abi.encodeParameters(
+      ['uint256[]', 'address[]', 'bytes', 'uint256[]'],
+      this.params
+    )
+    return keccak256(paramsEncoded) as Buffer
+  }
+
+  async json(): Promise<ProposalTransactionJSON[]> {
+    const contractDetails = await obtainKitContractDetails(this.kit)
+    const blockExplorer = new BlockExplorer(this.kit, contractDetails)
+
+    return concurrentMap(1, this.transactions, async (transaction) => {
+      const parsedTx = blockExplorer.tryParseTx(transaction as Transaction)
+      if (parsedTx == null) {
+        throw new Error(`Unable to parse ${transaction} with block explorer`)
+      }
+      return { callDetails: parsedTx.callDetails, value: parsedTx.tx.value }
+    })
+  }
+}
+
+type TxParams = Pick<ProposalTransaction, 'to' | 'value'>
+export class ProposalTransactionFactory {
```

Comment:
Just as a general design point (but not a blocking issue), this Java style use of Classes without an actual need for classes is not really typical in Javascript. If every method is static, why have a class at all? More common would be to just make functions and export them. @mcortesi WDYT?
Related talk I'd recommend: https://www.youtube.com/watch?v=QyJZzq0v7Z4